include/alpaka/atomic/AtomicOmpBuiltIn.hpp

```diff
@@ -0,0 +1,326 @@
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST AtomicOmpBuiltIn(AtomicOmpBuiltIn const &) = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST AtomicOmpBuiltIn(AtomicOmpBuiltIn &&) = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST auto operator=(AtomicOmpBuiltIn const &) -> AtomicOmpBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST auto operator=(AtomicOmpBuiltIn &&) -> AtomicOmpBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            /*virtual*/ ~AtomicOmpBuiltIn() = default;
+        };
+
+        namespace traits
+        {
+            //#############################################################################
+            //! The OpenMP accelerator atomic operation: ADD
+            template<
+                typename T,
+                typename THierarchy>
+            struct AtomicOp<
+                op::Add,
+                atomic::AtomicOmpBuiltIn,
+                T,
+                THierarchy>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_ACC_CUDA_ONLY static auto atomicOp(
+                    atomic::AtomicOmpBuiltIn const &,
+                    T * const addr,
+                    T const & value)
+                -> T
+                {
+                    T old;
+                    auto & ref(*addr);
+                    // atomically update ref, but capture original value in old
+                    #pragma omp atomic capture
+                    { old = ref; ref += value; }
+                    return old;
+                }
+            };
+
+            //#############################################################################
+            //! The OpenMP accelerator atomic operation: SUB
+            template<
+                typename T,
+                typename THierarchy>
+            struct AtomicOp<
+                op::Sub,
+                atomic::AtomicOmpBuiltIn,
+                T,
+                THierarchy>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_ACC_CUDA_ONLY static auto atomicOp(
+                    atomic::AtomicOmpBuiltIn const &,
+                    T * const addr,
+                    T const & value)
+                -> T
+                {
+                    T old;
+                    auto & ref(*addr);
+                    // atomically update ref, but capture original value in old
+                    #pragma omp atomic capture
+                    { old = ref; ref -= value; }
```

Comment:
Isn't such formatting odd? I mean, i see the logic, but find it not worth it (subjective ofc).